htdocs/core/tpl/card_presend.tpl.php

```diff
@@ -167,6 +167,27 @@
 	);
 	complete_substitutions_array($substitutionarray, $outputlangs, $object, $parameters);
 
+	// Find the good contact adress
+	$custcontact = '';
+	$contactarr = array();
+	$contactarr = $object->liste_contact(- 1, 'external');
+
+	if (is_array($contactarr) && count($contactarr) > 0) {
+		foreach ($contactarr as $contact) {
+			if (in_array($contact['code'], array('BILLING'))) {
```

Comment:
Why the __CONTACTCIVNAME__ is the billing address. It could be also the commercial contact, the proposal contact, the delivery contact.
So i suggest to introduce other substitutions key like:
__CONTACT_NAME_XXX__
And the contact use is the one with code XXX

Reply:
I think it's a good way to set contact substitution key